cscs-checks/mch/src/set_openacc_cuda_mpi.F90

```diff
@@ -0,0 +1,147 @@
+! This code tests MPI tasks communication with GPU devices 
+! using OpenACC directives and setting one device per task
+program set_openacc_cuda_mpi
+  use openacc
+  implicit none
+
+  include 'mpif.h'
+#ifdef CRAY
+  integer, parameter :: ACC_DEVICE_TYPE = 8
+#else
+  integer, parameter :: ACC_DEVICE_TYPE = 4
+#endif
+  integer, parameter :: ARRAYSIZE = 10
+  integer(kind=ACC_DEVICE_TYPE) :: devicetype
+  integer :: status(MPI_STATUS_SIZE), mpi_size, mpi_rank
+  integer :: ierr, i, gpuid, ngpus, localsum(2), globalsum(2)
+  real, allocatable :: array1(:), array2(:) 
+  
+  call MPI_Init(ierr)
+  call MPI_Comm_size(MPI_COMM_WORLD, mpi_size, ierr)
+  call MPI_Comm_rank(MPI_COMM_WORLD, mpi_rank, ierr)
+
+! each task creates two different arrays: the sum of their elements will be 10*mpi_rank
+  allocate(array1(ARRAYSIZE))
+  allocate(array2(ARRAYSIZE))
+
+#ifdef _OPENACC
+  if (mpi_rank == 0) then 
+    devicetype = acc_get_device_type()
+    ngpus = acc_get_num_devices(devicetype)
+    write(*,*) "MPI test with OpenACC using", mpi_size, "tasks and ", ngpus-1, "GPU devices"
```

Comment:
also in cosmo the mapping is such the the last tasks use no GPU, but I assume this should not matter.
So that n=0 uses GPU=0, n=1 uses GPU=1 , n=8 use no GPU